scripts/rpm/post-install.sh

```diff
@@ -48,41 +24,15 @@ if [[ ! -f /etc/telegraf/telegraf.conf ]] && [[ -f /etc/telegraf/telegraf.conf.s
    cp /etc/telegraf/telegraf.conf.sample /etc/telegraf/telegraf.conf
 fi
 
+# Set up log directories
+LOG_DIR=/var/log/telegraf
 test -d $LOG_DIR || mkdir -p $LOG_DIR
 chown -R -L telegraf:telegraf $LOG_DIR
 chmod 755 $LOG_DIR
 
-# Distribution-specific logic
-if [[ -f /etc/redhat-release ]] || [[ -f /etc/SuSE-release ]]; then
-    # RHEL-variant logic
-    if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
-        install_systemd /usr/lib/systemd/system/telegraf.service
-    else
-        # Assuming SysVinit
-        install_init
-        # Run update-rc.d or fallback to chkconfig if not available
-        if which update-rc.d &>/dev/null; then
-            install_update_rcd
-        else
-            install_chkconfig
-        fi
-    fi
-elif [[ -f /etc/os-release ]]; then
-    source /etc/os-release
-    if [[ "$NAME" = "Amazon Linux" ]]; then
-        # Amazon Linux 2+ logic
-        install_systemd /usr/lib/systemd/system/telegraf.service
-    elif [[ "$NAME" = "Amazon Linux AMI" ]]; then
-        # Amazon Linux logic
-        install_init
-        # Run update-rc.d or fallback to chkconfig if not available
-        if which update-rc.d &>/dev/null; then
-            install_update_rcd
-        else
-            install_chkconfig
-        fi
-    elif [[ "$NAME" = "Solus" ]]; then
-        # Solus logic
-        install_systemd /usr/lib/systemd/system/telegraf.service
-    fi
+# Set up systemd service
+if [[ -d /run/systemd/system ]]; then
+    cp -f /usr/lib/telegraf/scripts/telegraf.service /usr/lib/systemd/system/telegraf.service
+    systemctl enable telegraf || true
+    systemctl daemon-reload || true
```

Comment:
I noticed that the `|| true` was there before so maybe we should leave it, but it doesn't seem right to ignore the return value of these calls. Also, I don't think the script is running with `set -e` so it shouldn't fail even if they don't return 0.

Do you think it's worth removing the `|| true`?

Reply:
agreed on removing